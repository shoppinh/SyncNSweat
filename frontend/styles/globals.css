/* Basic global styles for Sync & Sweat frontend */
body {
  margin: 0;
  font-family: 'Segoe UI', 'Roboto', 'Arial', sans-serif;
  background: #f7f7fa;
  color: #222;
}
a {
  color: #0070f3;
  text-decoration: none;
}
a:hover {
  text-decoration: underline;
}
button {
  background: #0070f3;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1rem;
}
button:hover {
  background: #005bb5;
}
