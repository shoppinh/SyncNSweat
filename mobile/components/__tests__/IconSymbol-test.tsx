import { describe, expect, it, jest } from '@jest/globals';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { IconSymbol } from '../ui/IconSymbol';

// Mock MaterialIcons
jest.mock('@expo/vector-icons/MaterialIcons', () => 'MaterialIcons');

describe('<IconSymbol />', () => {
  it('renders correctly with required props', () => {
    const tree = renderer.create(
      <IconSymbol name="house.fill" color="#000000" />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('applies custom size', () => {
    const { getByTestId } = render(
      <IconSymbol 
        name="house.fill" 
        color="#000000" 
        size={32} 
        testId="icon-symbol"
      />
    );
    
    const icon = getByTestId('icon-symbol');
    expect(icon.props.size).toBe(32);
  });

  it('applies custom style', () => {
    const customStyle = { margin: 10 };
    
    const { getByTestId } = render(
      <IconSymbol 
        name="house.fill" 
        color="#000000" 
        style={customStyle} 
        testId="icon-symbol"
      />
    );
    
    const icon = getByTestId('icon-symbol');
    expect(icon.props.style).toEqual(customStyle);
  });

  it('maps SFSymbol names to MaterialIcons names', () => {
    const { getByTestId } = render(
      <IconSymbol 
        name="house.fill" 
        color="#000000" 
        testId="icon-symbol"
      />
    );
    
    const icon = getByTestId('icon-symbol');
    // house.fill should map to 'home' for MaterialIcons
    expect(icon.props.name).toBe('home');
  });
});
