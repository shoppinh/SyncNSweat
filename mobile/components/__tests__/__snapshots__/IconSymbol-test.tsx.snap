// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconSymbol /> renders correctly with required props 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="house.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="#000000"
  type="monochrome"
  weight="regular"
/>
`;
